shotgun_api3/shotgun.py

```diff
@@ -270,6 +270,7 @@ def __init__(self):
         self.session_token = None
         self.authorization = None
         self.no_ssl_validation = False
+        self.multi_entity_update_modes = ['set','add','remove']
```

Comment:
My thinking here was that by specifying client-side, we could:

- Be more explicit as to which values are accepted.
- Do validation checks before reaching the server, effectively reducing time to failure and adding redundancy.
- Provide an error message that's very specific to these modes. (Although arguably we should just do better error messaging for this case server-side anyway!)
- The likelihood of modes being updated frequency is low.

While it does seem there are some benefits client-side, I do agree this code should be as generic as possible. Will update.